prow/tide/tide.go

```diff
@@ -493,26 +493,49 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 
 // isPassingTests returns whether or not all contexts set on the PR except for
 // the tide pool context are passing.
-func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest) bool {
+func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest, bp config.TideBranchProtection) bool {
 	log = log.WithFields(pr.logFields())
 	contexts, err := headContexts(log, ghc, &pr)
 	if err != nil {
 		log.WithError(err).Error("Getting head commit status contexts.")
 		// If we can't get the status of the commit, assume that it is failing.
 		return false
 	}
+
+	requiredChecksSet := sets.NewString()
+	if bp.UseRequiredCheckOnly {
+		requiredChecks, err := ghc.GetRequiredChecks(
```

Comment:
This could use too many API tokens. We need to cache this info, but I'm not sure what a sane eviction policy would be. Maybe just use a 1 hour TTL?
A proxy cache layer would handle this transparently. @stevekuznetsov @BenTheElder

Reply:
I don't really like the idea that we'll pull this from GitHub and lag behind the current state or that we'll instead eat API tokens for this. We could accomplish the same thing by instead requiring you to set the list of required statuses in Prow when you set them on GitHub.

It's more effort and but it's also much less complex and less magical. This worked fine for the submit-queue...

Reply:
With the branch protector we're also currently deriving these from Prow config for Kubernetes repos anyhow, that would just leave non-Prow statuses, which generally seem to be made required without automation anyhow.